crates/sui-graphql-rpc/src/types/transaction_block_effects.rs

```diff
@@ -46,79 +54,111 @@ impl TransactionBlockEffects {
 
     /// Whether the transaction executed successfully or not.
     async fn status(&self) -> Option<ExecutionStatus> {
-        Some(match self.native.status() {
-            NativeExecutionStatus::Success => ExecutionStatus::Success,
-            NativeExecutionStatus::Failure { .. } => ExecutionStatus::Failure,
+        let sucess = match self.effects {
+            Either::Left(ref native) => native.status().is_ok(),
+            Either::Right(ref pre_commit_effects) => pre_commit_effects.status().is_ok(),
+        };
+
+        Some(match sucess {
+            true => ExecutionStatus::Success,
+            false => ExecutionStatus::Failure,
         })
     }
 
     /// The latest version of all objects (apart from packages) that have been created or modified
     /// by this transaction, immediately following this transaction.
     async fn lamport_version(&self) -> u64 {
-        self.native.lamport_version().value()
+        match self.effects {
+            Either::Left(ref native) => native.lamport_version(),
+            Either::Right(ref pre_commit_effects) => pre_commit_effects.lamport_version(),
+        }
+        .value()
     }
 
     /// The reason for a transaction failure, if it did fail.
     async fn errors(&self) -> Option<String> {
-        match self.native.status() {
-            NativeExecutionStatus::Success => None,
-
-            NativeExecutionStatus::Failure {
-                error,
-                command: None,
-            } => Some(error.to_string()),
-
-            NativeExecutionStatus::Failure {
-                error,
-                command: Some(command),
-            } => {
-                // Convert the command index into an ordinal.
-                let command = command + 1;
-                let suffix = match command % 10 {
-                    1 => "st",
-                    2 => "nd",
-                    3 => "rd",
-                    _ => "th",
-                };
-
-                Some(format!("{error} in {command}{suffix} command."))
+        let status = match self.effects {
```

Comment:
This change is just to keep uniformity between the error reprs.
Going from `ExecutionStatus` to `SuiExecutionStatus` condenses some info, so the reverse is non-trivial.
Since `SuiExecutionStatus` already contains a String error, we return that.
Open to other ideas.